### PR TITLE
Discussion: render intermediate materialization

### PR DESCRIPTION
**Pull request by schrockn**

### CODE REVIEW

File: python_modules/dagster/dagster/core/execution_plan/simple_engine.py [hunk 1]
```diff
@@ -301,7 +304,11 @@ def _create_step_output_event(step_context, step_output_value, intermediates_man
             value=value,
         )
 
-        return DagsterEvent.step_output_event(
+        yield DagsterEvent.step_materialization(
```
Comment (alangenfeld) [inline python_modules/dagster/dagster/core/execution_plan/simple_engine.py]: i think we should only do this when we are persisting intermediates somewhere and it should probably be its own event type.

File: python_modules/dagster/dagster_tests/core_tests/execution_plan_tests/test_execution_plan_subset.py [hunk 2]
```diff
@@ -41,15 +41,15 @@ def test_execution_plan_simple_two_steps():
 
     step_events = execute_plan(execution_plan)
     # start, out, success, start, out, success
```
Comment (alangenfeld) [inline python_modules/dagster/dagster_tests/core_tests/execution_plan_tests/test_execution_plan_subset.py]: [1]

File: python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_execution_plan.py [hunk 3]
```diff
@@ -174,14 +174,14 @@ def test_successful_one_part_execute_plan(snapshot):
 
     step_events = query_result['stepEvents']
     # 0-2 are sum_solid.num.input_thunk
```
Comment (alangenfeld) [inline python_modules/dagster-graphql/dagster_graphql_tests/graphql/test_execution_plan.py]: [1]

**Comment by schrockn** [comment]
pursuing this direction with a different approach here https://dagster.phacility.com/D13

**Pull request closed without merging**